src/problem.ts

```diff
@@ -1,62 +1,77 @@
-import { httpObject, ProblemInterface } from "types";
+import {
+  httpObject,
+  ProblemInterface,
+  ToJsonParamType,
+  UpdateProblemParamType,
+} from "./types";
 
-enum COPY_MODE {
-    SKIP_PROPS = 'skipProps',
-    LEAVE_PROPS = 'leaveProps'
+export enum COPY_MODE {
+  SKIP_PROPS = "skipProps",
+  LEAVE_PROPS = "leaveProps",
 }
 
 export class Problem extends Error implements ProblemInterface {
-    public type: string;
-    public title: string;
-    public instance?: string;
-    public detail?: string;
-    public http?: httpObject
-    [key: string]: any;
-
-
-    constructor(problem: ProblemInterface, customKeys?: string[]) {
-        super(problem.detail || problem.title);
-        this.http = problem.http
-        this.type = problem.type
-        this.title = problem.title;
-        this.detail = problem.detail;
-        this.instance = problem.instance;
-        this.stack = problem.stack;
-        customKeys?.map((customKey) => {
-            this[customKey] = problem[customKey];
-        })
-    }
-
-   copy(problem: ProblemInterface, mode: COPY_MODE, props: string[]): Problem {
-        switch (mode) {
-
-            case COPY_MODE.LEAVE_PROPS:
-                return new Problem(problem, props);
-
-            case COPY_MODE.SKIP_PROPS:
-            default:
-                let keysToBeCopied: string[] = [];
-                for (let key in problem) {
-                    if (props.includes(key))
-                        continue;
-                    keysToBeCopied.push(key)
-                }
-                return new Problem(problem, keysToBeCopied)
-        }
-    };
+  public type: string;
+  public title: string;
+  public instance?: string;
+  public detail?: string;
+  public http?: httpObject;
+  [key: string]: any;
 
-   toJSON(problem: Problem, includeStack = false): ProblemInterface {
+  constructor(problem: ProblemInterface, customKeys?: string[]) {
+    super(problem.detail || problem.title);
+    this.http = problem.http;
+    this.type = problem.type;
+    this.title = problem.title;
+    this.detail = problem.detail;
+    this.instance = problem.instance;
+    this.stack = problem.stack;
+    customKeys?.map((customKey) => {
+      this[customKey] = problem[customKey];
+    });
+  }
```

Comment:
I don't think so that people should pass as second argument customKeys, we should check that on `problem` argument. Please fix that and base logic only on `problem` argument to copy custom keys without `customKeys` arg. Let me know with any problems.

Reply:
You are right, it will be difficult for existing applications to integrate the problem api. 
For example, suppose an app has 1000 different custom keys, so its not a good idea to have those passed separately.

I have re-implemented this logic to be handled, natively within the problem object without any passing of extra  parameters.

Thanks @magicmatatjahu for this. You can re-review the latest commit.